ringpop.thrift

```diff
@@ -0,0 +1,25 @@
+struct Change {
+    1: required string source;
+    2: required string address;
+    3: required string status;
+    4: required double incarnationNumber;
+}
+
+struct JoinResult {
+    1: required list<Change> changes;
+}
+
+struct PingResult {
+    1: required list<Change> changes;
+}
+
+struct PingReqResult {
+    1: required bool pingStatus;
+    2: required list<Change> changes;
+}
+
+service Ringpop {
+    JoinResult join(1:string app, 2:string source, 3:double incarnationNumber);
+    PingResult ping(1:double checksum, 2:list<Change> changes);
+    PingReqResult pingReq(1:double checksum, 2:list<Change> changes, 3:string target);
+}
```

Comment:
if this is an opportunity to define network interface, do we want to further simply it?

just some thoughts
- gossip layer join/sync and ping operation may reuse the same network rpc to ping and exchange membership/updates
- add meta field to Change so that we could disseminate meta/config/app updates as well
- make responses having same structure

```
struct Change {
    1: required string source;
    2: required string address;
    3: required string status;
    4: required double incarnationNumber;
    5: required string meta;
}

struct Ack {
    1: required bool ok;
    2: required list<Change> changes;
}

service Ringpop {
    Ack ping(1:double checksum, 2:list<Change> changes);
    Ack pingReq(1:double checksum, 2:list<Change> changes, 3:string target);
}
```